iceoryx2-ffi/cxx/include/iox2/service_builder_request_response.hpp

```diff
@@ -140,9 +153,135 @@ class ServiceBuilderRequestResponse {
 
     void set_parameters();
 
-    iox2_service_builder_pub_sub_h m_handle = nullptr;
+    template <typename PayloadType>
+    auto get_payload_type_name() -> internal::FromCustomizedPayloadTypeName<PayloadType>;
+
+    template <typename PayloadType>
+    auto get_payload_type_name() -> internal::FromNonSlice<PayloadType>;
+
+    template <typename PayloadType>
+    auto get_payload_type_name() -> internal::FromSliceWithCustomizedInnerPayloadTypeName<PayloadType>;
+
+    template <typename PayloadType>
+    auto get_payload_type_name() -> internal::FromSliceWithoutCustomizedInnerPayloadTypeName<PayloadType>;
+
+    template <typename UserHeaderType>
+    auto get_user_header_type_name() ->
+        typename std::enable_if_t<internal::HasUserHeaderTypeNameMember<UserHeaderType>::value, const char*>;
+
+    template <typename UserHeaderType>
+    auto get_user_header_type_name() ->
+        typename std::enable_if_t<!internal::HasUserHeaderTypeNameMember<UserHeaderType>::value, const char*>;
+
+    iox2_service_builder_request_response_h m_handle = nullptr;
 };
 
+template <typename RequestPayload,
+          typename RequestHeader,
+          typename ResponsePayload,
+          typename ResponseHeader,
+          ServiceType S>
+template <typename PayloadType>
+inline auto ServiceBuilderRequestResponse<RequestPayload, RequestHeader, ResponsePayload, ResponseHeader, S>::
```

Comment:
I think all type name translation functions that are shared with publish subscribe should be:
1. Moved out of the publish subscribe builder
2. Renamed so that they are generic and depend only on the actual payload type or header type
3. Shall be used in the publish subscriber and request response builder

When we implement the blackboard pattern, we have to do again a type translation and when we implement it then a third time it gets really error prone.